Automattic/wp-calypso
# Add GDPR location refresh to avoid missing GA page view

#### Changes proposed in this Pull Request

##### Abstract

Without a user logged in  loading Calypso will not register a page view initially ( perhaps depending on the speed of network requests ) since the request to `public-api.wordpress.com/geo` will not return in time to OK the tracking. This is affecting the Jetpack tracking situation especially hard since we have a cross-domain flow of 

`jetpack.com` -> `cloud.jetpack.com/pricing` -> `wordpress.com/checkout`

which will be affected by this issue twice.

##### Changes

* Make sure we await a GDPR county code before disallowing Google Analytics functions
* Update `refreshCountryCodeCookieGdpr` to prevent duplicate requests while one is already running
*
#### Testing instructions



1. set `ad-tracking` to true in `jetpack-cloud-development.json`, and boot calypso green on trunk ( not the branch yet )
2. navigate to `jetpack.cloud.localhost:3000/pricing` in an incognito window, and enter `localStorage.setItem( 'debug', 'calypso:analytics:ga' );` into the dev tools console
3. delete the `county_code` cookie and refresh the page
4. verify `calypso:analytics:ga [Disallowed] analytics recordPageView(` appears in your console
5. switch to this branch
6. delete the `county_code` cookie and refresh the page
7. verify `calypso:analytics:ga Recording Page View ~ [URL: /pricing] ` appears in your console
